libs/SmartSync/SmartSync/Classes/Target/SFMruSyncDownTarget.m

```diff
@@ -126,7 +126,16 @@ - (void) getListOfRemoteIds:(SFSmartSyncSyncManager*)syncManager
                         from:self.objectType]
                        whereClause:inPredicate]
                       build];
-    [self startFetch:syncManager maxTimeStamp:0 queryRun:soql errorBlock:errorBlock completeBlock:completeBlock];
+
+    SFSyncDownTargetFetchCompleteBlock fetchBlock = ^(NSArray* records) {
+        NSMutableArray * remoteIds = [NSMutableArray new];
+        for (NSDictionary * record in records) {
```

Comment:
get remote ids should really "return" (meaning through the complete block) the ids instead of records.